ethkey/src/brain_recover.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashSet;
+
+use edit_distance::edit_distance;
+use parity_wordlist;
+
+use super::{Address, Error, Brain, Generator};
+
+pub fn brain_recover(
+	address: &Address,
+	known_phrase: &str,
+	expected_words: usize,
+	filter: Option<(usize, usize)>,
+) -> Result<String, Error> {
+	let known_words = parity_wordlist::WORDS.iter().cloned().collect::<HashSet<_>>();
+	let mut words = known_phrase.split(' ')
+		.map(|word| match known_words.get(word) {
+			None => {
+				info!("Invalid word '{}', looking for potential substitutions.", word);
+				let substitutions = generate_substitutions(word);
```

Comment:
what if I defined words myself?

Reply:
Then we can't help you. This recovery process is targeted at people who used parity-generated phrase but made a mistake while backing it up.
We could make a more generic recovery process (which allows you to provide your own dictionary) as an extra feature.